glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -534,8 +536,9 @@ open class GleanInternalAPI internal constructor () {
      * file.
      *
      * @param pingNames List of ping names to submit.
+     * @param reason The reason the ping is being sent.
      */
-    internal fun submitPingsByNameSync(pingNames: List<String>) {
+    internal fun submitPingsByNameSync(pingNames: List<String>, reason: String? = null) {
```

Comment:
Please note that `pingNames` can potentially contain names of different pings (e.g. `baseline`, `events`) and we're only passing a single reason: we should probably have a list of pairs or a map, or two lists in order to match specific optional reasons to specific ping names.

Reply:
This is a good point, and I agree this is a problem.  Passing in "parallel lists", one of which is optional, seems kind of a strange API though (at least it's internal API).

I actually feel like it might be better to remove all of the plural submit "pings" in favor of a single submit "ping" on all of these functions.  We only use this to submit multiple pings in one place, and we don't even get any parallelism out of thing by batching them up like this.  I think that would be a useful simplification.

Thoughts?  @Dexterp37, @badboy, @travis79, @brizental ?

Reply:
I guess, specifically, what I'd propose is to merge this PR as-is with respect to this, and as follow-on work, replace `submitPings*` with `submitPing`, which would fix the ambiguity around reason codes.

Reply:
One other reason we went with `submitPings` (multiple) was to only call into Rust once.
So far no one has ever complained about FFI calls and given how infrequent and with how small of a list we call it I'm fine with dropping that API

(Users can only submit one ping at a time anyway, through the `CustomPing.submit()` function, right?)

Reply:
I'm all for the one at a time, one ping with one reason passed into the function.  I also like the idea of this simplifying the underlying FFI function.

Reply:
@badboy: Yes, though the public API, users can only submit one ping at once.  The only place where we submit multiple is on going to background (to send baseline and events).

Reply:
I'm fine with dropping the plural, but I think it needs to be part of this PR, since this makes #687 buggy (unless we opt for sending the ping on background without a "reason" code.) :-)

Reply:
Ah, good point about #687.  I was mainly just trying to not make this PR any bigger than it already is.  But it should be straightforward.  I'll take it on.